kubespawner/spawner.py

```diff
@@ -3048,14 +3068,39 @@ async def _load_profile(self, slug, selected_profile_user_options):
             # are in the form data posted. This prevents users who may be authorized
             # to only use one profile from being able to access options set for other
             # profiles
-            for user_selected_option_name in selected_profile_user_options.keys():
-                if (
-                    user_selected_option_name
-                    not in profile.get('profile_options').keys()
-                ):
-                    raise ValueError(
-                        f'Expected option {user_selected_option_name} for profile {profile["slug"]}, not found in posted form'
-                    )
+
+            for option_name, option in profile.get('profile_options').items():
+                if option_name not in selected_profile_user_options:
+                    # other_choice in Enabled:
+                    if option.get('other_choice', {}).get('enabled', False):
+                        if (
+                            f'{option_name}--other-choice'
```

Comment:
Given `{option_name}--other-choice` is used multiple times, can we make it into a variable for clarity? `other_choice_form_key` or something?

Reply:
I agree.

The problem here is that we still need to define the variable in two places, since we have these two different loops. So, I had to create the variable twice: https://github.com/jupyterhub/kubespawner/pull/735/commits/f55e32302a9e80a967c9531fbf10068ca6d7970f#diff-6fdb3a928956e8ab2b810679d51fae72870d45ed4a088e85e16814339aff8901R3074 and https://github.com/jupyterhub/kubespawner/pull/735/commits/f55e32302a9e80a967c9531fbf10068ca6d7970f#diff-6fdb3a928956e8ab2b810679d51fae72870d45ed4a088e85e16814339aff8901R3107 .

This feels really not nice. I think options:

 - Make get_other_choice_key a function or so and call the function from both places so at least we define the `--other-choice` constant in only one place.
 - Refactor to not use two loops and loop only once
 - Leave as is in my latest commit
 - Something else .. ?

Thanks much for the review!